pmd-core/src/main/java/net/sourceforge/pmd/lang/document/FileCollector.java

```diff
@@ -277,7 +335,7 @@ public boolean addDirectory(Path dir) throws IOException {
             reporter.error("Not a directory {0}", dir);
             return false;
         }
-        Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+        Files.walkFileTree(dir, EnumSet.of(FileVisitOption.FOLLOW_LINKS), Integer.MAX_VALUE, new SimpleFileVisitor<Path>() {
```

Comment:
That's the change, that will now follow symlinks, if someone provides a symlink as `--dir`... should we do this change?

Reply:
I think so, yes